### PR TITLE
[prometheus-logstash] add logstash exporter

### PR DESCRIPTION
**Pull request by ebuildy**
Signed-off-by: Thomas Decaux <ebuildy@gmail.com>

#### What this PR does / why we need it:

An exporter for logstash

#### Checklist

- [X] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [X] Chart Version bumped
- [X] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/prometheus-logstash-exporter/README.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+_See [helm repo](https://helm.sh/docs/helm/helm_repo/) for command documentation._
+
+## Install Chart
+
+```console
+# Helm 3
+$ helm install [RELEASE_NAME] prometheus-community/prometheus-logstash-exporter
+
+# Helm 2
+$ helm install --name [RELEASE_NAME] prometheus-community/prometheus-logstash-exporter
```
Comment (monotek) [inline charts/prometheus-logstash-exporter/README.md:30]: helm 2 is deprecated. i would not mention it any longer.

File: charts/prometheus-logstash-exporter/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+apiVersion: v1
+description: Logstash stats exporter for Prometheus
+name: prometheus-logstash-exporter
+version: 0.1.1
```
Comment (monotek) [inline charts/prometheus-logstash-exporter/Chart.yaml:4]: why 0.1.1 and not 0.1.0 as semver suggest for initial version?
see: https://semver.org/#how-should-i-deal-with-revisions-in-the-0yz-initial-development-phase

File: charts/prometheus-logstash-exporter/README.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+_See [helm install](https://helm.sh/docs/helm/helm_install/) for command documentation._
+
+## Uninstall Chart
+
+```console
+# Helm 3
+$ helm uninstall [RELEASE_NAME]
+
+# Helm 2
```
Comment (monotek) [inline charts/prometheus-logstash-exporter/README.md:45]: helm 2 is deprecated. i would not mention it any longer.

File: charts/prometheus-logstash-exporter/README.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+```
+
+This removes all the Kubernetes components associated with the chart and deletes the release.
+
+_See [helm uninstall](https://helm.sh/docs/helm/helm_uninstall/) for command documentation._
+
+## Upgrading Chart
+
+```console
+# Helm 3 or 2
```
Comment (monotek) [inline charts/prometheus-logstash-exporter/README.md:56]: helm 2 is deprecated. i would not mention it any longer.

File: charts/prometheus-logstash-exporter/README.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+_See [helm upgrade](https://helm.sh/docs/helm/helm_upgrade/) for command documentation._
+
+## Configuration
+
+See [Customizing the Chart Before Installing](https://helm.sh/docs/intro/using_helm/#customizing-the-chart-before-installing).
+To see all configurable options with detailed comments, visit the chart's [values.yaml](./values.yaml), or run these configuration commands:
+
+```console
+# Helm 2
```
Comment (monotek) [inline charts/prometheus-logstash-exporter/README.md:68]: helm 2 is deprecated. i would not mention it any longer.

File: charts/prometheus-logstash-exporter/ci/default-values.yaml [hunk 6]
```diff
@@ -0,0 +1 @@
+# Leave this file empty to ensure that CI runs builds against the default configuration in values.yaml.
```
Comment (monotek) [inline charts/prometheus-logstash-exporter/ci/default-values.yaml:1]: imho the file is not needed if empty.
at least i thought the default values are always checked?

File: charts/prometheus-logstash-exporter/templates/deployment.yaml [hunk 7]
```diff
@@ -0,0 +1,147 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ template "logstash-exporter.fullname" . }}
+  labels:
+    chart: {{ template "logstash-exporter.chart" . }}
```
Comment (monotek) [inline charts/prometheus-logstash-exporter/templates/deployment.yaml:6]: please use the more recent labeling defaults which are using templating.

you can copy it from the chart created by:

```
helm create prometheus-logstash-exporter
```

File: charts/prometheus-logstash-exporter/values.yaml [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+# To use a service account not handled by the chart, set the name here
+# and set create to false
+serviceAccount:
+  create: false
+  name: default
+
+# Creates a PodSecurityPolicy and the role/rolebinding
+# allowing the serviceaccount to use it
+podSecurityPolicies:
+  enabled: false
```
Comment (monotek) [inline charts/prometheus-logstash-exporter/values.yaml:126]: any reason this is disabled?
imho should be enabled as long as the defined values work out of the box.

File: charts/prometheus-logstash-exporter/Chart.yaml [hunk 9]
```diff
@@ -0,0 +1,17 @@
+apiVersion: v1
```
Comment (monotek) [inline charts/prometheus-logstash-exporter/Chart.yaml:1]: helm 2 is deprecated so use v2 here.

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/prometheus-logstash-exporter/ci/default-values.yaml (hunk 6 above)
Comment (ebuildy) [reply charts/prometheus-logstash-exporter/ci/default-values.yaml:1]: right, I copy it from elasticsearch chart, will remove it

File: charts/prometheus-logstash-exporter/values.yaml (hunk 8 above)
Comment (ebuildy) [reply charts/prometheus-logstash-exporter/values.yaml:126]: copied from elasticsearch-exporter chart, I am changing it

**Comment by monotek** [comment]
@ebuildy  
Could you fix the linter errors please?

**Review by monotek — CHANGES_REQUESTED** [review]
@ebuildy please fix linter errors.

**Comment by TsofiaTsuriel1234** [comment]
Is this PR is ready for merge? when are you going to merge it?
Thanks!

**Comment by monotek** [comment]
No, it's not ready.

The DCO check fails: https://github.com/prometheus-community/helm-charts/pull/628/checks?check_run_id=1904680196
Pleas fix.

@prometheus-community/helm-charts-admins 
The Codeowners check needs approval of one of the repo owners: https://github.com/prometheus-community/helm-charts/pull/628/checks?check_run_id=1904680517

**Review by monotek — CHANGES_REQUESTED** [review]
@TsofiaTsuriel1234 
No, it's not ready.

The DCO check fails: https://github.com/prometheus-community/helm-charts/pull/628/checks?check_run_id=1904680196

@ebuildy  please fix.

@prometheus-community/helm-charts-admins
The Codeowners check needs approval of one of the repo owners: https://github.com/prometheus-community/helm-charts/pull/628/checks?check_run_id=1904680517

**Comment by oded-dd** [comment]
@ebuildy Any update on that? Great!

**Comment by pwlawe** [comment]
@monotek We have a need for this, I'd like to revive this work.  I'll make a new PR and reference this one.  FYI @ebuildy

**Comment by monotek** [comment]
Could you please do this?
https://github.com/prometheus-community/helm-charts/pull/628/checks?check_run_id=3666839649

**Pull request closed without merging**